lib/drivers/insuletDriver.js

```diff
@@ -1512,7 +1520,15 @@ module.exports = function (config) {
       postrecords = buildBGRecords(data, postrecords);
       // first sort by log index
       postrecords = _.sortBy(postrecords, function(d) {
-        return d.index;
+        if (d.index) {
+          return d.index;
+        }
+        else if (d.payload && d.payload.logIndices) {
+          return d.payload.logIndices[0];
+        }
+        else {
```

Comment:
Ah damn, let me look at this some more. It appears the `undefined`s are being sorted to the end by lodash, but I don't want that - these are non-log records (certain alarms). Let me play around a bit.